### PR TITLE
TST: Skip db tests unless explicitly specified in -m pattern

### LINKED ISSUE #24485: TST: psycopg2 errors when running tests locally
I just started noticing these (167 of them) in the logs from `python -m pytest pandas/tests --skip-slow`

```
_____________________________ ERROR at setup of TestPostgreSQLAlchemyConn.test_read_sql[load_iris_data0] ______________________________

cls = <pandas.tests.io.test_sql.TestPostgreSQLAlchemyConn object at 0x7f2843ab4a50>

    @pytest.fixture(autouse=True, scope='class')
    def setup_class(cls):
        cls.setup_import()
        cls.setup_driver()
        conn = cls.connect()
>       conn.connect()

[...]
>       conn = _connect(dsn, connection_factory=connection_factory, async=async)
E       OperationalError: (psycopg2.OperationalError) fe_sendauth: no password supplied

../../.local/lib/python2.7/site-packages/psycopg2/__init__.py:164: OperationalError
```

### PR DESCRIPTION
**Pull request by datapythonista**
- [x] closes #24485

Since #24450 we don't skip the db tests if the connection fails. That's useful, because the CI won't fail silently if there is any problem with the databases. But as reported in #24485, it's common to have `pymysql` or `psycopg2` installed locally, but no database connection. This is making the tests fail locally, and the agreed solution is to skip the db tests unless they are explicitly requested. 

But pytest doesn't make it easy to skip tests by default when using `-m`, but this solution should work. Another option would be an `--include-db` option, but I think it makes reading the expressions in the build very difficult (like, what `--only-slow --include-db` runs?). With `-m` the expressions are much clearer (e.g. `slow and db`), so I think this option is better.

The `eval` is how `pytest` itself evaluates the value of the `-m` parameter.

### CODE REVIEW

**Comment by gfyoung** [comment]
@datapythonista : I'm confused by the motivation for your changes here...

Where have we defined a flag for `--include-db` ?

File: pandas/conftest.py [hunk 1]
```diff
@@ -54,14 +54,23 @@ def pytest_runtest_setup(item):
...
-    if 'db' in item.keywords and item.config.getoption("--skip-db"):
-        pytest.skip("skipping due to --skip-db")
-
     if 'high_memory' in item.keywords and not item.config.getoption(
             "--run-high-memory"):
         pytest.skip(
             "skipping high memory test since --run-high-memory was not set")
 
+    # if "db" not explicitly set in the -m pattern, we skip the db tests
+    import collections
```
Comment (jreback) [inline pandas/conftest.py]: can you move the import to the top

**Review by jreback — CHANGES_REQUESTED** [review]
minor comment, otherwise lgtm.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. merge on green.

**Comment by jreback** [comment]
thanks!

**Pull request merged**